From dec1118a28d19fb620e45bd673433240a357c24e Mon Sep 17 00:00:00 2001
From: Frank Wunderlich <frank-w@public-files.de>
Date: Thu, 13 Nov 2025 19:20:21 +0100
Subject: [PATCH 067/116] arm64: dts: mediatek: mt7988a-bpi-r4: add overlay to
 disable pcie2

Some Modems like FM350-GL require pcie disabled on m.2 key-b slot (CN16)
to work properly.

Signed-off-by: Frank Wunderlich <frank-w@public-files.de>
---
based on overlay file from user Andrew
https://forum.banana-pi.org/t/fm350-gl-and-be14-finally-works-on-bpi-r4/19170/40
---
 arch/arm64/boot/dts/mediatek/Makefile          |  1 +
 .../mt7988a-bananapi-bpi-r4-nopcie2.dtso       | 18 ++++++++++++++++++
 2 files changed, 19 insertions(+)
 create mode 100644 arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-nopcie2.dtso

diff --git a/arch/arm64/boot/dts/mediatek/Makefile b/arch/arm64/boot/dts/mediatek/Makefile
index a548a9802832..637027cd20fc 100644
--- a/arch/arm64/boot/dts/mediatek/Makefile
+++ b/arch/arm64/boot/dts/mediatek/Makefile
@@ -24,6 +24,7 @@ dtb-$(CONFIG_ARCH_MEDIATEK) += mt7986b-rfb.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-2g5.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-emmc.dtbo
+dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-nopcie2.dtbo
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-pro-4e.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-pro-8x.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt7988a-bananapi-bpi-r4-pro-cn15.dtbo
diff --git a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-nopcie2.dtso b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-nopcie2.dtso
new file mode 100644
index 000000000000..4ff61d984446
--- /dev/null
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-nopcie2.dtso
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright (C) 2023 MediaTek Inc.
+ * Author: Frank Wunderlich <frank-w@public-files.de>
+ */
+
+/dts-v1/;
+/plugin/;
+
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "bananapi,bpi-r4", "mediatek,mt7988a";
+};
+
+&{/soc/pcie@11280000} {
+	status = "disabled";
+};
-- 
2.39.5

